storage/store/inmemory/unsynchronized/transactions.go

```diff
@@ -0,0 +1,44 @@
+package unsynchronized
+
+import (
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+type Transactions struct {
+	//TODO: we don't need a mutex here as we have a guarantee by design
+	// that we write data only once and it happens before the future reads.
+	// We decided to leave a mutex for some time during active development.
+	// It'll be removed later on.
+	lock  sync.RWMutex
+	store map[flow.Identifier]*flow.TransactionBody
+}
+
+func NewTransactions() *Transactions {
+	return &Transactions{
+		store: make(map[flow.Identifier]*flow.TransactionBody),
+	}
+}
+
+var _ storage.Transactions = (*Transactions)(nil)
+
+func (t *Transactions) ByID(txID flow.Identifier) (*flow.TransactionBody, error) {
+	t.lock.RLock()
+	val, ok := t.store[txID]
+	t.lock.RUnlock()
+
+	if !ok {
+		return nil, storage.ErrNotFound
+	}
+
+	return val, nil
+}
+
+func (t *Transactions) Store(tx *flow.TransactionBody) error {
+	t.lock.Lock()
+	defer t.lock.Unlock()
+	t.store[tx.ID()] = tx
```

Comment:
the persisted db will skip duplicate inserts, we should replicate that here as well

https://github.com/onflow/flow-go/blob/6a4e69be1a80023e4a4815663b5010a751cf6bfb/storage/badger/transactions.go#L21-L23